pkg/server/server.go

```diff
@@ -24,22 +24,34 @@ type Service interface {
 // Server represents a bundle of services defined by a specific ID.
 // Examples of servers are the satellite, the storagenode, and the uplink.
 type Server struct {
-	lis      net.Listener
-	grpc     *grpc.Server
-	next     []Service
-	identity *identity.FullIdentity
+	publicListener  net.Listener
+	privateListener net.Listener
+	grpc            *grpc.Server
```

Comment:
You'll need to add public grpc / private grpc separately.

One nice way to organize these is to have:

```
public struct {
	listener net.Listener
	grpc     *grpc.Server
}
private struct {
	listener net.Listener
	grpc     *grpc.Server
}
```